legacy/dockerregistry/inventory.go

```diff
@@ -1593,6 +1594,9 @@ func MkReadRepositoryCmdReal(
 ) stream.Producer {
 	var sh stream.HTTP
 
+	// Log GCR query (only if --verbose flag is present)
+	logclient.CountGCRQuery(1)
```

Comment:
Can you just increment a single global variable here (guard it with Mutex lock/unlock)?

Reply:
That is exactly what this function does. However, I need this global variable and mutex to exist somewhere. After attempting to place it in legacy/audit/auditor.go, legacy/cli/audit.go, and legacy/audit/types.go, I was continually met with a cyclical dependency problem. This occurs because there are many files that need to touch this mutex and global variable. Therefore, I opted to create a group of Public function as an extension of the logclient package.

Reply:
TLDR: I can't just create a mutex and global var because many files require importing this logic.

Reply:
Ideally this should be abstracted away as part of the logger. i.e. I should not need to know the incrementing logic inside `inventory.go`

This would look something like someLogger.Debug("Querying GCR with blah blah")
and depending on whether it is a `PersistentLogger`, `NormalLogger`, `NoopLogger` etc etc different thing willl happen
incrementing should be done automatically by the `PersistentLogger`.